lib/resty/rediscluster.lua

```diff
@@ -58,62 +86,135 @@ end
 _M.parse_key = parse_key
 
 
-local slot_cache = {}
-local master_nodes = {}
-
-local cmds_for_all_master = {
-    ["flushall"] = true,
-    ["flushdb"] = true
-}
-
-local cluster_invalid_cmds = {
-    ["config"] = true,
-    ["shutdown"] = true
-}
-
 local function redis_slot(str)
     return redis_crc(parse_key(str))
 end
 
-local function check_auth(self, redis_client)
-    -- redis before 6.0
-    if type(self.config.auth) == "string" then
+
+local function release_connection(red, config)
+    local ok,err = red:set_keepalive(config.keepalive_timeout
+            or DEFAULT_KEEPALIVE_TIMEOUT, config.keepalive_cons or DEFAULT_KEEPALIVE_CONS)
+    if not ok then
+        ngx_log(NGX_ERR, "set keepalive failed: ", err)
+    end
+end
+
+
+local check_auth
```

Comment:
just curiosity, why use closures here?

Reply:
The enclosed `check_reused` is only used by the `check_auth`, so try to pack them together.